tests/unit/client.py

```diff
@@ -0,0 +1,659 @@
+# Copyright 2013-2014 Massachusetts Open Cloud Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the
+# License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS
+# IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied.  See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Unit tests for client library"""
+import haas
+from haas import model, api, deferred, server, config
+from haas.model import db
+from haas.network_allocator import get_network_allocator
+import pytest
+import json
+import requests
+import os
+import tempfile
+import subprocess
+import time
+from subprocess import check_call, Popen
+from urlparse import urljoin
+import requests
+from requests.exceptions import ConnectionError
+from haas.client.base import ClientBase
+from haas.client.client import Client, RequestsHTTPClient, KeystoneHTTPClient
+
+
+ep = "http://127.0.0.1:8000" or os.environ.get('HAAS_ENDPOINT')
+username = "hil_user" or os.environ.get('HAAS_USERNAME')
+password = "hil_pass1234" or os.environ.get('HAAS_PASSWORD')
+
+http_client = RequestsHTTPClient()
+http_client.auth = (username, password)
+C = Client(ep, http_client)  # Initializing client library
+MOCK_SWITCH_TYPE = 'http://schema.massopencloud.org/haas/v0/switches/mock'
+OBM_TYPE_MOCK = 'http://schema.massopencloud.org/haas/v0/obm/mock'
+OBM_TYPE_IPMI = 'http://schema.massopencloud.org/haas/v0/obm/ipmi'
+
+
+class Test_ClientBase:
+    """ When the username, password is not defined
+    It should raise a LookupError
+    """
+
+    def test_init_error(self):
+        try:
+            x = ClientBase()
+        except TypeError:
+            assert True
+
+# FIX ME: The test may vary based on which backend session is used.
+#    def test_correct_init(self):
+#        x = ClientBase(ep, 'some_base64_string')
+#        assert x.endpoint == "http://127.0.0.1:8000"
+#        assert x.auth == "some_base64_string"
+
+    def test_object_url(self):
+        x = ClientBase(ep, 'some_base64_string')
+        y = x.object_url('abc', '123', 'xy23z')
+        assert y == 'http://127.0.0.1:8000/abc/123/xy23z'
+
+# For testing the client library we need a running HIL server, with dummy
+# objects populated. Following classes accomplish that end.
+# It shall:
+#       1. Configures haas.cfg
+#       2. Instantiates a database
+#       3. Starts a server on an arbitary port
+#       4. Populates haas with dummy objects
+#       5. tears down the setup in a clean fashion.
+
+
+# pytest.fixture(scope="module")
```

Comment:
`make_config` is not a fixture. `create_setup` is a fixture that calls both make_config and `fin` calls `cleanup`.